Environment variables are evaluated as integer instead of string

I got this error:

```
registry_1    | goroutine 1 [running]:
registry_1    | panic(0xb9b860, 0xc420012400)
registry_1    | 	/usr/local/go/src/runtime/panic.go:500 +0x1a1
registry_1    | github.com/docker/distribution/registry/handlers.NewApp(0x10657c0, 0xc4203f9470, 0xc4203c2a80, 0x10657c0)
registry_1    | 	/go/src/github.com/docker/distribution/registry/handlers/app.go:123 +0x3908
registry_1    | github.com/docker/distribution/registry.NewRegistry(0x7fbc34e92040, 0xc4203f9470, 0xc4203c2a80, 0xc4203c2a80, 0x0, 0x0)
registry_1    | 	/go/src/github.com/docker/distribution/registry/registry.go:86 +0x213
registry_1    | github.com/docker/distribution/registry.glob..func1(0x108d140, 0xc4203ee640, 0x1, 0x1)
registry_1    | 	/go/src/github.com/docker/distribution/registry/registry.go:55 +0x106
registry_1    | github.com/docker/distribution/vendor/github.com/spf13/cobra.(*Command).execute(0x108d140, 0xc4203ee5f0, 0x1, 0x1, 0x108d140, 0xc4203ee5f0)
registry_1    | 	/go/src/github.com/docker/distribution/vendor/github.com/spf13/cobra/command.go:495 +0x190
registry_1    | github.com/docker/distribution/vendor/github.com/spf13/cobra.(*Command).Execute(0x108d2e0, 0xc4201eff40, 0xc4200001a0)
registry_1    | 	/go/src/github.com/docker/distribution/vendor/github.com/spf13/cobra/command.go:560 +0x3c3
registry_1    | main.main()
registry_1    | 	/go/src/github.com/docker/distribution/cmd/registry/main.go:24 +0x2d
```

the reason is I have set the ENV Variable `REGISTRY_STORAGE_SWIFT_TENANT=12341234`

Now the registry code interprets the value as an int instead of an string.

I also have run into this problem. Here is how to reproduce:

1) Set your `OS_TENANT` environment variable to an all numeric value. Then start `registry:2` like this:

```
docker run -d \
  --name registry \
  --publish 127.0.0.1:5000:5000 \
  --restart always \
  -e REGISTRY_STORAGE=swift \
  -e REGISTRY_STORAGE_SWIFT_REGION=${OS_REGION} \
  -e REGISTRY_STORAGE_SWIFT_TENANT=${OS_TENANT} \
  -e REGISTRY_STORAGE_SWIFT_USERNAME=${OS_USERNAME} \
  -e REGISTRY_STORAGE_SWIFT_PASSWORD=${OS_PASSWORD} \
  -e REGISTRY_STORAGE_SWIFT_AUTHURL=${OS_AUTH_URL} \
  -e REGISTRY_STORAGE_SWIFT_CONTAINER=docker-registry \
  registry:2
```

If you run `docker logs registry` after that point, you should see the panic message. Example:

```
panic: 1 error(s) decoding:

* 'Tenant' expected type 'string', got unconvertible type 'int'

goroutine 1 [running]:
panic(0xb9b860, 0xc42036bde0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/docker/distribution/registry/handlers.NewApp(0x10657c0, 0xc420344bd0, 0xc4202a5180, 0x10657c0)
	/go/src/github.com/docker/distribution/registry/handlers/app.go:123 +0x3908
github.com/docker/distribution/registry.NewRegistry(0x7f6f82d0e198, 0xc420344bd0, 0xc4202a5180, 0xc4202a5180, 0x0, 0x0)
	/go/src/github.com/docker/distribution/registry/registry.go:86 +0x213
github.com/docker/distribution/registry.glob..func1(0x108d140, 0xc420332270, 0x1, 0x1)
	/go/src/github.com/docker/distribution/registry/registry.go:55 +0x106
github.com/docker/distribution/vendor/github.com/spf13/cobra.(*Command).execute(0x108d140, 0xc420332220, 0x1, 0x1, 0x108d140, 0xc420332220)
	/go/src/github.com/docker/distribution/vendor/github.com/spf13/cobra/command.go:495 +0x190
github.com/docker/distribution/vendor/github.com/spf13/cobra.(*Command).Execute(0x108d2e0, 0xc42019df40, 0xc4200001a0)
	/go/src/github.com/docker/distribution/vendor/github.com/spf13/cobra/command.go:560 +0x3c3
main.main()
	/go/src/github.com/docker/distribution/cmd/registry/main.go:24 +0x2d
panic: 1 error(s) decoding:

* 'Tenant' expected type 'string', got unconvertible type 'int'
```

--- a/registry/storage/driver/swift/swift.go
+++ b/registry/storage/driver/swift/swift.go
@@ -142,6 +142,19 @@ func FromParameters(parameters map[string]interface{}) (*Driver, error) {
 		InsecureSkipVerify: false,
 	}
 
+	// Sanitize some entries before trying to decode parameters with mapstructure
+	// TenantID and Tenant when integers only and passed as ENV variables
+	// are considered as integer and not string. The parser fails in this
+	// case.
+	_, ok := parameters["tenant"]
+	if ok {
+		parameters["tenant"] = fmt.Sprint(parameters["tenant"])
+	}
+	_, ok = parameters["tenantid"]
+	if ok {
+		parameters["tenantid"] = fmt.Sprint(parameters["tenantid"])
+	}
+
 	if err := mapstructure.Decode(parameters, &params); err != nil {
 		return nil, err
 	}

